### DISCUSSION TITLE
Significance of "#define USER_SETUP_ID 70" in User_Setup file?

### DISCUSSION THREAD

**Discussion by JonRobert** [Bodmer/TFT_eSPI#2619] [Q&A - General]
I am using an ESP32C3 with a ST7796s display.   I've created a User_Setup.h based on the "Setup70c_ESP32_C3_ILI9341.h" file in the library.  See below.

My question is the line
`#define USER_SETUP_ID 70`
from the original "Setup70c_ESP32_C3_ILI9341.h" file.  What is the significance of this define?  Now that I'm not using the "Setup70c_ESP32_C3_ILI9341.h" file but a modified copy named "User_Setup.h"  what if anything do I need to define in place of the questioned line?

I don't have hardware results as I am waiting for my display to be delivered, so I'm trying to get as much ready as I can.

Thanks
John

most of my User_Setup.h file:   (sorry I couldn't get the <> to work with multiple lines)

`// Setup for the ESP32 C3 with ILI9341 display`

// Note SPI DMA with ESP32 C3 is not currently supported

#define USER_SETUP_ID 70
// See SetupX_Template.h for all options available
//#define ILI9341_DRIVER
#define ST7796_DRIVER
//#define ILI9488_DRIVER
//ESP32 C3 XIAO
#define TFT_CS    4
#define TFT_MOSI 10
#define TFT_MISO  9
#define TFT_SCLK  8
#define TFT_DC    2
#define TFT_RST   3
`

**Comment by Bodmer** [comment]
#define USER_SETUP_ID 70 just allows your sketch to print out diagnostics or check if the correct setup file is being used:
```
Serial.println(USER_SETUP_ID);
```
It can be useful if you have multiple hardware setups and forget to change the library reference.

**Comment by JonRobert** [comment]
Thank you for your answer, and for you library. 
John
